common/macresman.cpp

```diff
@@ -606,39 +606,41 @@ void MacResManager::readMap() {
 	}
 }
 
-String MacResManager::constructAppleDoubleName(String name) {
+Path MacResManager::constructAppleDoubleName(Path name) {
 	// Insert "._" before the last portion of a path name
-	for (int i = name.size() - 1; i >= 0; i--) {
+	String rawName = name.rawString();
```

Comment:
Might be overkill, but how about adding functions to Common::Path that output something like Common::List<Common::String> pathComponents, and a similar function for creating a Common::Path from such a list?

That would make it easier to transform each path-component, without having to care about the DIR_SEPARATOR directly.